src/main/java/com/ontotext/kafka/GraphDBSinkTask.java

```diff
@@ -26,32 +21,46 @@
  */
 public class GraphDBSinkTask extends SinkTask {
 
-	private static final Logger LOG = LoggerFactory.getLogger(GraphDBSinkTask.class);
-	private Repository repository;
-	private final AtomicBoolean shouldRun = new AtomicBoolean(true);
-	private final ConcurrentLinkedQueue<Collection<SinkRecord>> sinkRecords = new ConcurrentLinkedQueue<>();
+	private static final Logger log = LoggerFactory.getLogger(GraphDBSinkTask.class);
+
+	/**
+	 * Use a single processor and queue for all tasks of a single unique connector.
+	 */
+	private static final Map<String, SinkRecordsProcessor> processors = new HashMap<>();
 	private GraphDBSinkConfig config;
-	private SinkRecordsProcessor recordProcessor;
-	private Thread recordProcessorThread;
+	private SinkRecordsProcessor processor;
 
 
 	@Override
-	public String version() {
-		return VersionUtil.getVersion();
+	public void start(Map<String, String> properties) {
+		this.config = new GraphDBSinkConfig(properties);
+		log.info("Starting the GraphDB sink task for connector {}", config.getConnectorName());
+		this.processor = startProcessor(config);
+		log.info("Configuration complete.");
 	}
 
-	@Override
-	public void start(Map<String, String> properties) {
-		LOG.info("Starting the GraphDB sink task");
-		config = new GraphDBSinkConfig(properties);
-		repository = initializeRepository(repository, config);
-		LOG.info("Initialized GraphDB repository connection: repository {}, GraphDB instance {}", config.getRepositoryId(),
-			config.getServerUrl());
-		recordProcessor = createProcessor();
-		recordProcessorThread = new Thread(recordProcessor);
-		shouldRun.set(true);
-		recordProcessorThread.start();
-		LOG.info("Configuration complete.");
+	/**
+	 * Creates a new {@link SinkRecordsProcessor} and starts its thread, for a corresponding connector name.
+	 * If a context has already been created (i.e. from another task initialization for the same connector), return the existing queue, without starting
+	 * any new threads.
+	 *
+	 * @param config - The configuration for which to create the new context
+	 * @return ProcessorContext
+	 */
+	private SinkRecordsProcessor startProcessor(GraphDBSinkConfig config) {
+		String connectorName = config.getConnectorName();
+		if (processors.containsKey(connectorName)) {
+			return processors.get(connectorName);
+		}
+		synchronized (processors) {
```

Comment:
Can't this synchronized block be simplified if Concurrent Hash Map is used instead?

Reply:
Concurrent Hash Map's read op, i.e. get() is not blocking, i.e. one thread can read old data (while another is updating it). Therefore, if two tasks start at the same time (I am unaware if tasks are started in sequence or in parallel and this is up to implementation so beyond our concern) , they would both be able to see there is no connector processor, so they will both create a processor instance. This is what this synchronization tries to avoid.